enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+roles.
+
+#### Use of metal3
+
+[metal3](https://metal3.io/) will be used to boot the assisted discovery ISO
+for bare-metal deployments. Specifically, the BareMetalHost API will be used
+from a Hub cluster to boot the discovery ISO on hosts that will be used to
+form new clusters.
+
+For this first iteration, it is assumed that the hub cluster itself is using
+the bare metal platform, and thus will have baremetal-operator installed
+and available for use. In the future it may be desirable to make the metal3
+capabilities available on hub clusters that are not using the bare metal
+platform.
+
+#### BareMetalHost can boot live ISOs
+
+A separate [enhancement to
+metal3](https://github.com/metal3-io/metal3-docs/pull/150) proposes a new
+capability in the BareMetalHost API enabling it to boot live ISOs other than
+the Ironic Python Agent. That feature is required so that automation in a
+cluster can boot the discovery ISO on known hardware as the first step toward
+provisioning that hardware.
+
+#### Host Approval
+
+It is important that when an agent checks in, it not be allowed to join a
+cluster until an actor has approved it. From a security standpoint, it is not
+OK for anyone who can access a copy of a discovery ISO and/or access the API
+where agents report themselves to implicitly have the capability to join a
+cluster as a Node.
+
+The Agent CRD will have a field in which to designate that it is approved. If
+the host was booted via the baremetal-operator, approval will be granted
+automatically. The same automation that caused the host to boot would have the
+ability to recognize the resulting Agent and mark it as approved.
+
+#### Day 2 Add Node Boot-it-Yourself
+
+#### Personas for multi-cluster management
+
+**Infra Owner** Manages physical infrastructure and configures hosts to boot
+the discovery ISO that runs the Agent.
+
+**Cluster Creator** Uses running Agents to create and grow clusters.
+
+#### Create Cluster
+
+This scenario takes place on a hub cluster where hive and possibly RHACM are
+present. This scenario does not include centralized machine management.
+
+This scenario is an end-to-end flow that enables the user to specify everything
+up-front and then have an automated process create a cluster. Because the
+user's primary frame of reference is hardware-oriented, this flow enables them
+to specify Node attributes such as "role" with their BareMetalHost definition.
+Specifying those attributes together is particularly useful in a gitops
+approach.
+
+1. Infra Owner creates an InstallEnv resource. It can include fields such as egress proxy, NTP server, SSH public key, ... In particular it includes an Agent label selector, and a separate field of labels that will be automatically applied to Agents.
+1. Infra Owner creates BareMetalHost resources with corresponding BMC credentials. They must be labeled so that they match the selctor on the InstallEnv and must be in the same namespace as the InstallEnv.
```

Comment:
Why do we need both the labels and the namespacing?

Reply:
There could be multiple InstallEnv resources in the same namespace.

Reply:
What's the use case for that? It feels like it increases the support matrix quite a bit.

Which of the fields in the env resource would we expect to vary for 2 instances in the same namespace, and why?

I see that the resource today has a single field for a cluster reference. Does having multiple env resources in a namespace imply we would have multiple clusters as well?

Reply:
There are two use cases driving that. They are not use cases we are aiming to solve with this proposal, but we need to leave the door open to supporting them in the near future. That's why they're not listed in the "Use Cases" section, but maybe we can find somewhere to state these considerations more clearly.
1. Given a single InstallEnv, its hardware could be used to form multiple cluster.
2. When creating or expanding a single cluster, it can make sense to utilize hardware from multiple InstallEnvs, particularly when an InstallEnv represents a remote location (for remote worker nodes) or distinct availability zones.

Based on that, I think we're best off assuming that one namespace can contain multiple clusters and multiple install environments.

The single cluster reference on an InstallEnv is a temporary requirement until we can support creating ISOs in the back-end that are not cluster-specific.